### PR TITLE
Deleting students should result in deleting the corresponding lessons

### PR DESCRIPTION
**Pull request by Ma-Xizi**

### CODE REVIEW

**Comment by Ma-Xizi** [comment]
The tests will be refined after merging the previous PRs:)

File: src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java [hunk 1]
```diff
@@ -52,12 +54,23 @@ public CommandResult execute(Model model) throws CommandException {
...
+        deleteLessonsForStudent(model, personToDelete);
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, format(personToDelete)));
     }
 
+    private void deleteLessonsForStudent(Model model, Person student) {
+        // Collect lessons associated with the student
+        List<Lesson> lessonsToDelete = model.getFilteredLessonList().stream()
+                .filter(lesson -> lesson.getStudent().equals(student))
+                .collect(Collectors.toList());
```
Comment (Ma-Xizi) [inline src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java:68]: This part is the most important:)

File: src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java [hunk 2]
```diff
@@ -52,12 +54,23 @@ public CommandResult execute(Model model) throws CommandException {
...
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        deleteLessonsForStudent(model, personToDelete);
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, format(personToDelete)));
     }
 
+    private void deleteLessonsForStudent(Model model, Person student) {
+        // Collect lessons associated with the student
+        List<Lesson> lessonsToDelete = model.getFilteredLessonList().stream()
```
Comment (NorbertLoh) [inline src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java:66]: Should we find lessons to delete from the entire list? As we are trying to delete everything.

File: src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java [hunk 3]
```diff
@@ -52,12 +54,23 @@ public CommandResult execute(Model model) throws CommandException {
...
         }
-
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        deleteLessonsForStudent(model, personToDelete);
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, format(personToDelete)));
     }
 
+    private void deleteLessonsForStudent(Model model, Person student) {
```
Comment (NorbertLoh) [inline src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java:64]: It is also possible to have this method in the LessonSchedule.java such that the method takes in a person and will delete it within itself and we do not need to extract the list from the class. Therefore we just have to call deleteLessonsOfPerson(person).

**Review by NorbertLoh — COMMENTED** [review]
Nits to fix. Otherwise LGTM!

**Comment by Ma-Xizi** [comment]
> Nits to fix. Otherwise LGTM!

Just realized that ethan has alr written deleteStudentLesson in the modelManager class and I dunno need to reimplement it.

**Review by NorbertLoh — APPROVED** [review]
LGTM!

**Pull request closed without merging**